lib/server-app.js

```diff
@@ -65,6 +65,29 @@ proto.middlewareFromConfig = function(factory, config) {
   }
 
   var handler = factory.apply(null, params);
+
+  // Check if methods/verbs filter exists
+  var verbs = config.methods || config.verbs;
+  if (Array.isArray(verbs)) {
+    if (verbs.indexOf('all') === -1) {
+      var originalHandler = handler;
+      if (handler.length <= 3) {
+        handler = function(req, res, next) {
+          if (verbs.indexOf(req.method.toLowerCase()) === -1) {
```

Comment:
> For a tiny array, indexOf() should be very performed.

Fair enough.

How about converting the method/verbs values specified by the config to lowercase? I think it's more likely that a user writes `POST` instead of `post` in the configuration, compared to  `req.method` being upper-case.